discovery_child_development/notebooks/04_train_initial_model.py

```diff
@@ -0,0 +1,444 @@
+    # make a copy, otherwise this will modify the original dataframe even though we are not returning it
+    df_copy = df.copy()
+
+    # Convert each row into a string representation
+    df_copy["combined"] = df_copy.apply(
+        lambda row: "".join(row.astype(str).values), axis=1, result_type="reduce"
+    )
+
+    # Find the most common set of binary labels
+    most_common_set = df_copy["combined"].value_counts().idxmax()
+    count = df_copy["combined"].value_counts().max()
+
+    return most_common_set, count
+
+
+def flatten_classification_report(report):
+    """
+    Flatten a scikit learn classification report into a format that can be stored on wandb
+    """
+    flat_report = {}
+    for class_label, metrics in report.items():
+        if class_label not in ["micro avg", "macro avg", "weighted avg", "samples avg"]:
+            for metric, value in metrics.items():
+                flat_report[f"class_{class_label}_{metric}"] = value
+        else:
+            flat_report[class_label] = report[class_label]
+    return flat_report
+
+
+# %%
+# Initialize a run and log score threshold with wandb
+run = wandb.init(
+    project="ISS supervised ML",
+    job_type="Baseline modeling",
+    save_code=True,
+    config={"score_threshold": SCORE_THRESHOLD},
+)
+
+# %%
+CONCEPT_IDS.replace("|", "_")
+
+# %%
+# Load data from s3
+training_data_filename = (
+    f"openalex_data_{CONCEPT_IDS}_year-2019-2020-2021-2022-2023_train.csv"
+)
+# We will use set the wandb description to be the dataset name (which includes details of concepts and years)
+# - we set the description, and not the name, as there is a limit on the length of artifact names.
+# wandb artifact names also cannot include "|" so we replace this.
+wandb_name = training_data_filename.replace("|", "_")
+
+openalex_data = S3.download_obj(
+    S3_BUCKET,
+    path_from=f"{INPUT_PATH}{training_data_filename}",
+    download_as="dataframe",
+    kwargs_reading={"index_col": 0},
+)
+
+# add reference to this data in wandb
+wb.add_ref_to_data(
+    run, "openalex_train_data_raw", wandb_name, S3_BUCKET, training_data_filename
+)
+
+openalex_data.head()
+
+# %% [markdown]
+# ## EDA
+
+# %% [markdown]
+# Check how many OpenAlex IDs there are in the data (~20,000)
+
+# %%
+len(openalex_data["openalex_id"].unique())
+
+# %% [markdown]
+# Check distributions of `level` and `sub_category`. We can notice:
+# * It turns out most of the concepts/categories that we determined to be relevant were level 1 or level 2 concepts
+# * There are around 30,000 instances of something being tagged with "general development", which is more than the number of OpenAlex works!! That means that some of the concepts from within the "general development" category cover basically the whole dataset. In terms of creating a baseline model, we could try just predicting that every single paper is 1 for "general development" and 0 for all the other sub-categories, and see what evaluation metrics that gets.
```

Comment:
Yes, I guess general development will be our whole subset. Agree that we could train a general classifier that predicts whether text is about child development or not - then you'd need to collect a random subset of other, non-development papers from OpenAlex for the negative samples.